### PR TITLE
[Enhance] Add compatibility document

### PR DESCRIPTION
**Pull request by Wuziyi616**
I imitate the compatibility document in [MMDet](https://github.com/open-mmlab/mmdetection/blob/master/docs/compatibility.md) and write a (very initial and very naive) version. Only includes the change on ScanNet dataset.

### CODE REVIEW

**Comment by Wuziyi616** [comment]
Maybe we should add more terms to it, e.g. the model weight conversion for VoteNet on ScanNet?

File: docs/compatibility.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Compatibility with Previous Versions of MMDetection3D
+
+This document provides detailed descriptions of the BC-breaking changes in MMDetection3D.
+
+## MMDetection3D 0.13.0
+
+### ScanNet data pre-processing
+
+We adopt new pre-processing and conversion steps of ScanNet dataset. In previous versions of MMDetection3D, ScanNet dataset was only used for 3D detection task, where we trained on train set and tested on val set. In MMDetection3D 0.13.0, we further support 3D segmentation task on ScanNet, which includes online benchmarking on test set. Since the alignment matrix is not provided for test set data, we abandon the alignment of points in data generation step to support both tasks. Besides, as 3D segmentation requires per-point prediction, we also remove the down-sampling step in data generation.
```
Comment (Tai-Wang) [inline docs/compatibility.md:9]: trained on the training set and tested on the validation set.

Comment (Tai-Wang) [reply docs/compatibility.md:9]: we abandon the alignment of points in data generation step -> steps

File: docs/compatibility.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+- In our new ScanNet processing scripts, we save the unaligned points for all train, val and test set. For train and val set with annotations, we also store the `axis_align_matrix` in data infos. For ground-truth bounding boxes, we store boxes in both aligned and unaligned coordinates with key `gt_boxes_upright_depth` and key `unaligned_gt_boxes_upright_depth` respectively in data infos.
```
Comment (Tai-Wang) [inline docs/compatibility.md:11]: for all the training, validation and test set.

File: docs/compatibility.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+- In `ScanNetDataset`, we now load the `axis_align_matrix` as a part of data annotations. If it is not contained in old data infos, we will use identity matrix for compatibility. We also add a transform function `GlobalAlignment` in ScanNet detection data pipeline to align the points.
+
+- Since the aligned boxes share the same key as in old data infos, we do not need to modify the code related to it. But do remember that they are not in the same coordinate as the saved points.
+
+- There is a `IndoorPointSample` step in the data pipeline of ScanNet detection task which down-samples points. So the removal of down-sampling in data generation will not affect the code.
```
Comment (Tai-Wang) [inline docs/compatibility.md:17]: a -> an

File: docs/compatibility.md (hunk 1 above)
Comment (Wuziyi616) [reply docs/compatibility.md:9]: Fix.

File: docs/compatibility.md (hunk 2 above)
Comment (Wuziyi616) [reply docs/compatibility.md:11]: Fix.

File: docs/compatibility.md (hunk 3 above)
Comment (Wuziyi616) [reply docs/compatibility.md:17]: Fix.

**Review by Tai-Wang — APPROVED** [review]

**Comment by Wuziyi616** [comment]
One question: should we upload new pre-trained weights for models trained on newly processed ScanNet? I tried VoteNet and get a better performance compared to reported on the benchmark page.

File: docs/compatibility.md [hunk 4]
```diff
@@ -2,6 +2,22 @@
...
+
+- In our new ScanNet processing scripts, we save the unaligned points for all the training, validation and test set. For train and val set with annotations, we also store the `axis_align_matrix` in data infos. For ground-truth bounding boxes, we store boxes in both aligned and unaligned coordinates with key `gt_boxes_upright_depth` and key `unaligned_gt_boxes_upright_depth` respectively in data infos.
+
+- In `ScanNetDataset`, we now load the `axis_align_matrix` as a part of data annotations. If it is not contained in old data infos, we will use identity matrix for compatibility. We also add a transform function `GlobalAlignment` in ScanNet detection data pipeline to align the points.
+
+- Since the aligned boxes share the same key as in old data infos, we do not need to modify the code related to it. But do remember that they are not in the same coordinate as the saved points.
```
Comment (ZwwWayne) [inline docs/compatibility.md:15]: the same coordinate system

File: docs/compatibility.md [hunk 5]
```diff
@@ -2,6 +2,22 @@
 
 This document provides detailed descriptions of the BC-breaking changes in MMDetection3D.
 
+## MMDetection3D 0.14.0
+
+### ScanNet data pre-processing
+
+We adopt new pre-processing and conversion steps of ScanNet dataset. In previous versions of MMDetection3D, ScanNet dataset was only used for 3D detection task, where we trained on the training set and tested on the validation set. In MMDetection3D 0.14.0, we further support 3D segmentation task on ScanNet, which includes online benchmarking on test set. Since the alignment matrix is not provided for test set data, we abandon the alignment of points in data generation steps to support both tasks. Besides, as 3D segmentation requires per-point prediction, we also remove the down-sampling step in data generation.
+
+- In our new ScanNet processing scripts, we save the unaligned points for all the training, validation and test set. For train and val set with annotations, we also store the `axis_align_matrix` in data infos. For ground-truth bounding boxes, we store boxes in both aligned and unaligned coordinates with key `gt_boxes_upright_depth` and key `unaligned_gt_boxes_upright_depth` respectively in data infos.
```
[the remark this answers is not in the document]
Comment (Wuziyi616) [reply docs/compatibility.md:11]: Fix.

File: docs/compatibility.md (hunk 4 above)
Comment (Wuziyi616) [reply docs/compatibility.md:15]: Fix.

File: docs/compatibility.md [hunk 6]
```diff
@@ -2,6 +2,22 @@
...
+
+- In `ScanNetDataset`, we now load the `axis_align_matrix` as a part of data annotations. If it is not contained in old data infos, we will use identity matrix for compatibility. We also add a transform function `GlobalAlignment` in ScanNet detection data pipeline to align the points.
+
+- Since the aligned boxes share the same key as in old data infos, we do not need to modify the code related to it. But do remember that they are not in the same coordinate as the saved points.
+
+- There is an `IndoorPointSample` step in the data pipeline of ScanNet detection task which down-samples points. So the removal of down-sampling in data generation will not affect the code.
+
+We have trained a [VoteNet](https://github.com/open-mmlab/mmdetection3d/blob/master/configs/votenet/votenet_8x8_scannet-3d-18class.py) model on the newly processed ScanNet dataset and get similar benchmark results. In order to prepare ScanNet data for both detection and segmentation tasks, please re-run the new pre-processing scripts.
```
[the remark this answers is not in the document]
Comment (Wuziyi616) [reply docs/compatibility.md:19]: Add.
